src/Services/Caching/CacheService.cs

```diff
@@ -9,29 +10,99 @@ namespace AzureMcp.Services.Caching;
 public class CacheService(IMemoryCache memoryCache) : ICacheService
 {
     private readonly IMemoryCache _memoryCache = memoryCache;
+    private static readonly ConcurrentDictionary<string, HashSet<string>> _groupKeys = new();
 
-    public ValueTask<T?> GetAsync<T>(string key, TimeSpan? expiration = null)
+    public ValueTask<T?> GetAsync<T>(string group, string key, TimeSpan? expiration = null)
```

Comment:
group shold be optional right?

Reply:
No. It is required. It connects to the service. All clients using the same service should be in one group.

Reply:
I guess I'm not following why group would be required for all cache entries.

Reply:
Okay, it's fine as-is